### PR TITLE
ensure soft-deleted AddonUser instances are filtered out

### LINKED ISSUE #14681: Keep a permanent record of which users were an addon owner/developer
When a user is removed from an Addon we just remove it from the AddonUsers table, which means tracking down previous users means having to query via other data sources (logs, ActivityLog, etc).  

We should either soft-delete AddonUser, or add a further role of "Deleted" (and generally filter it out).  Either way needs to be done with care in regards to only exposing the non-deleted addonusers publicly, and handling a user being re-added back to an addon (which would effectively undelete).

### PR DESCRIPTION
**Pull request by eviljeff**
follow-up fix #14681 - I missed some places where we directly filtered on Addon/UserProfile directly.

### CODE REVIEW

File: src/olympia/amo/models.py [hunk 1]
```diff
@@ -521,16 +521,16 @@ def _get_manager_with_default_filtering(cls, manager, q_filter):
...
+                from_cache = (
+                    self.prefetch_cache_name in
+                    getattr(self.instance, '_prefetched_objects_cache', {}))
                 qs = super().get_queryset()
-                # Here is where we add the filter.
-                return qs.filter(q_filter or models.Q())
+                if not from_cache and q_filter:
+                    # Here is where we add the filter.
+                    qs = qs.filter(q_filter)
+                return qs
```
Comment (eviljeff) [inline src/olympia/amo/models.py:533]: Not _strictly_ a fix for the issue, but it made it clearer to me what the logic actually is.

File: src/olympia/devhub/tests/test_views.py [hunk 2]
```diff
@@ -57,6 +57,11 @@ def setUp(self):
         assert self.client.login(email='regular@mozilla.com')
         assert self.client.get(self.url).status_code == 200
         self.user_profile = UserProfile.objects.get(id=999)
+        not_their_addon = addon_factory(users=[user_factory()])
```
Comment (bobsilverberg) [inline src/olympia/devhub/tests/test_views.py:60]: Nit: It actually *is* their addon, isn't it? It's just that the `AddonUser` record has them listed as deleted.

File: src/olympia/devhub/feeds.py [hunk 3]
```diff
@@ -31,7 +30,7 @@ def items(self, key):
         if key.addon:
             addons = key.addon
         else:  # We are showing all the add-ons
-            addons = Addon.objects.filter(authors=key.user)
+            addons = key.user.addons.all()
```
Comment (bobsilverberg) [inline src/olympia/devhub/feeds.py:33]: Is there a test for this change? Is it covered by the change in ` src/olympia/devhub/tests/test_views.py`?

File: src/olympia/devhub/tests/test_views.py (hunk 2 above)
Comment (eviljeff) [reply src/olympia/devhub/tests/test_views.py:60]: It's not their add-on _any more_  😉 - they've been deleted from the addon owners.  (A user would never be added as deleted in reality)

File: src/olympia/devhub/feeds.py (hunk 3 above)
Comment (eviljeff) [reply src/olympia/devhub/feeds.py:33]: yeah, I added to the `setUp` for the base `HubTest` which is used by the feeds tests too.

**Pull request merged**